apps/common-app/src/examples/RuntimeTests/ReanimatedRuntimeTestsRunner/UpdatesContainer.ts

```diff
@@ -87,70 +84,110 @@ export function createUpdatesContainer(testRunner: TestRunner) {
+  ): [MultiViewSnapshot, 'multiView'] | [SingleViewSnapshot, 'singleView'] {
+    const updatesForTag: Record<number, Array<OperationUpdate>> = {};
+    for (const updateRequest of updates) {
+      const { tag } = updateRequest;
+
+      if (!(tag in updatesForTag)) {
+        updatesForTag[tag] = [];
       }
-    } else {
-      for (const updateRequest of jsUpdates.value) {
-        const filteredUpdate: Record<string, OperationUpdate> = {};
+      let update: OperationUpdate = [];
+      if (propsNames.length === 0) {
+        update = 'update' in updateRequest ? updateRequest.update : updateRequest.snapshot;
+      } else {
         for (const prop of propsNames) {
-          filteredUpdate[prop] = updateRequest.update[prop as keyof OperationUpdate];
+          update[prop] =
+            'update' in updateRequest
+              ? updateRequest.update[prop as keyof OperationUpdate]
+              : updateRequest.snapshot[prop as keyof OperationUpdate];
         }
-        updates.push(filteredUpdate);
       }
+      updatesForTag[tag].push(update);
+    }
+
+    const recordedMultipleViews = Object.keys(updatesForTag).length > 1;
+
+    if (recordedMultipleViews) {
+      return [updatesForTag, 'multiView'];
+    } else {
+      // In case of recording only one view return an array
+      const singleViewSnapshot: SingleViewSnapshot = updatesForTag[Number(Object.keys(updatesForTag)[0])];
+      return [singleViewSnapshot, 'singleView'];
+    }
+  }
+
+  function getUpdates(component?: TestComponent, propsNames: string[] = []): SingleViewSnapshot {
+    const [sortedUpdates, snapshotType] = _sortUpdatesByViewTag(jsUpdates.value, propsNames);
+    if (snapshotType === 'multiView') {
+      if (component === undefined) {
+        throw new Error('Recorded snapshots of many views, specify component you want to get snapshot of');
+      }
+      const tag = component?.getTag();
+      if (!tag || !(tag in sortedUpdates)) {
+        throw new Error('Snapshot of given component not found');
+      } else {
+        return sortedUpdates[tag];
+      }
+    } else {
+      return sortedUpdates;
     }
-    return updates;
   }
 
-  async function getNativeSnapshots(propsNames: string[] = []) {
+  async function getNativeSnapshots(component?: TestComponent, propsNames: string[] = []): Promise<SingleViewSnapshot> {
     const nativeSnapshotsCount = nativeSnapshots.value.length;
     const jsUpdatesCount = jsUpdates.value.length;
     if (jsUpdatesCount === nativeSnapshotsCount) {
       await testRunner.runOnUIBlocking(() => {
         'worklet';
         const lastSnapshot = nativeSnapshots.value[nativeSnapshotsCount - 1];
-        _makeNativeSnapshot(
-          [
-            {
-              tag: lastSnapshot.tag,
-              shadowNodeWrapper: lastSnapshot.shadowNodeWrapper,
-              update: lastSnapshot.snapshot,
-            },
-          ],
-          jsUpdatesCount - 1,
-        );
+        if (lastSnapshot) {
```

Comment:
When this is useful?

Reply:
When there is no last snapshot, because trying to access property of undefined yields error